### DISCUSSION TITLE
Neue Coding-Abenteuer: Pandas, Tic-Tac-Toe, NumPy und einen Wecker

### DISCUSSION THREAD

**Discussion by Roadraudi** [Roadraudi/Codespace_Repository#14] [Announcements]
Hey Leute! Hier sind meine neuesten Coding-Abenteuer, die ich gerne mit euch teilen möchte.

Pandas erkunden:
Ich habe mich kürzlich mit der Python-Bibliothek "Pandas" beschäftigt und meinen [Code](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/pandaUebung.py) um diese Datenanalyse-Bibliothek erweitert. Ich finde es hier besonders spannend was man eigentlich alles damit anstellen kann und habe deshalb mit unterschiedlichen Befehlen den Datenframe mir ausgeben lassen.

Tic-Tac-Toe Spiel fertiggestellt:
Ein weiteres spannendes Projekt war die Fertigstellung meines [Tic-Tac-Toe-Spiels](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/Tic-Tac-Toe.py). Es funktioniert zwar nur in der Konsole, aber die grundlegende Spiellogik ist implementiert. Spieler können abwechselnd ihre Züge machen, indem sie einfach die Nummer des Feldes angeben, in dem sie ihren Eintrag platzieren möchten.

Inspiration durch KI-Chatbots wie ChatGPT oder Copilot:
Ich habe auch KI-Chatbots wie ChatGPT genutzt, um mir einige kreative Ideen für meine Projekte zu holen. Sie sind wirklich hilfreich, um neue Lösungsansätze zu finden und Tipps zu bekommen, wie man Probleme angehen kann. Nicht um einfach den Code zu Kopieren sondern man kann seine Leistungsstärke individuell anpassen und konkret sich weiterhelfen lassen an bestimmten Stellen im Code. So bin ich auch auf mein nächstes Programm aufmerksam geworden und ich kann es euch nur ans Herz legen wenn ihr vielleicht auch ein paar mehr ausgefallene Programme coden möchtet.

Wecker Programm:
Schließlich habe ich ein [Wecker-Programm](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/Wecker/Wecker.py) entwickelt, das ihr in meinem [GIT-Repository](https://github.com/Roadraudi/Codespace_Repository/tree/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/Wecker) findet. Es enthält sogar eine Weck-Sounddatei. Ich bin gespannt, ob jemand den Mut hat, es als Wecker zu verwenden, und freue mich auf euer Feedback.

Neuestes [Programm](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Daily%20Coding/NumPy/numpyUebung.py) mit NumPy:
Heute habe ich mich zum ersten Mal mit NumPy beschäftigt und ein wenig damit herumgespielt. NumPy ist eine Python-Bibliothek, die das Arbeiten mit Arrays und Matrizen erleichtert. Ich habe einige grundlegende Operationen mit Arrays ausprobiert und finde die Möglichkeiten, die NumPy bietet, wirklich cool. Hier ist einmal ein [Tutorial](https://www.youtube.com/watch?v=XamW67vUJ-Q) auf Deutsch was ich ganz spannend fand und zudem habe ich mich noch auf der folgenden Seite: https://www.w3schools.com/python/numpy/default.asp näher damit beschäftigt.

Ich brauche eure Hilfe!
Ich habe eine Idee für ein weiteres Programm, nämlich ein Wetter Programm, eventuell sogar mit einer GUI. Das Problem dabei ist, dass ich keinen API-Schlüssel habe, um auf Wetterinformationen zuzugreifen. Diese Schlüssel sind häufig mit einem kostenpflichtigen Abo verbunden. Wenn also einer von euch einen Schlüssel hat oder Lust hat, mit mir dieses Projekt anzugehen, oder vielleicht sogar Tipps hat, wie man an Wetterinformationen kommt, soll sich bitte bei mir melden.

Bleibt dran für weitere Updates zu meinen Coding-Projekten! Bis bald!

**Comment by NicoBaier123** [comment]
Hey, deine Coding-Abenteuer klingen echt spannend! Pandas, Tic-Tac-Toe, Wecker-Programm - tolle Projekte! Wenn du Hilfe beim Wetter-Programm brauchst, melde dich bei mir. Weiter so!

**Comment by TheGhodra** [comment]
Das mit dem Wecker ist echt eine coole Idee und das Tic Tac Toe Game eauch, mach weiter so!
